Frontend/CarRentalAdmin/src/app/auth.service.ts

```diff
@@ -0,0 +1,52 @@
+import * as moment from "moment";
+import {Admin} from "./admin";
+import {Injectable} from "@angular/core";
+import {HttpClient} from "@angular/common/http";
+import {tap} from "rxjs/operators";
+import {Observable} from "rxjs";
+import {AuthResponse} from "./auth-responce";
+
+@Injectable()
+export class AuthService {
+  constructor(private http: HttpClient) {
+  }
+
+  login(admin: Admin): Observable<any> {
+    return this.http.post<AuthResponse>('/api/login', admin)
+      .pipe(
+        tap(
+          response => this.setSession(response)
+        )
+      )
+  }
+
+  private setSession(authResult: AuthResponse) {
+    const expiresAt = moment().add(authResult.expiresIn, 'second');
+
+    localStorage.setItem('id_token', authResult.idToken);
+    localStorage.setItem("expires_at", JSON.stringify(expiresAt.valueOf()) );
+  }
+
+  logout() {
+    localStorage.removeItem("id_token");
+    localStorage.removeItem("expires_at");
+  }
+
+  public isLoggedIn() {
+    return moment().isBefore(this.getExpiration());
+  }
+
+  public isLoggedOut() {
+    return !this.isLoggedIn();
+  }
+
+  getExpiration() {
+    const expiration = localStorage.getItem("expires_at");
+    if (expiration != null) {
+      const expiresAt = JSON.parse(expiration);
+      return moment(expiresAt);
+    }
+
+    return moment(0);
```

Comment:
```suggestion
    if (expiration == null) {
      return moment(0);
    }
    
    const expiresAt = JSON.parse(expiration);
    return moment(expiresAt);

```

it is a good practice to have positive condition in if statement (easier to keep in mind positive statement) and quit function as soon as it is clear that we have nothing to do in it (no need to scroll down and read useless for this case code)